### ISSUE TITLE
Bug: FS1 Training 5 does not proceed (Special training sexp)

### ISSUE THREAD

**Issue by bangstk** [scp-fs2open/fs2open.github.com#2825]
Hello;

In the current nightlies in FSPort, FS1 Training 5 (Advanced Training #2, or btm-05.fs2, or the one that teaches you about lockon missiles) is getting stuck.
When the instructor tells you to target Freighter 1, he's supposed to tell you to fire 1 Interceptor after you lock on.
But, instead, nothing happens, once you lock on the instructor says nothing and the training mission cannot proceed and cannot be completed. The game does not seem to be detecting that the player has locked onto Freighter 1.

This doesn't occur in the 19.0.0 stable release but is happening in the current nightlies, so I think it might be something the new multilock features switched around

This training mission uses the hardcoded special training sexps defined in process_special_sexps() in sexp,cpp, Case 0:
Here is the code it uses to check if player has achieved aspect lock on "Freighter 1"

```
case 0:	//	Ship "Freighter 1" is aspect locked by player.
		if (Player_ai->target_objnum != -1) {
			if (!(stricmp(Ships[Objects[Player_ai->target_objnum].instance].ship_name, "Freighter 1"))) {
				if (Player_ai->current_target_is_locked)
					return SEXP_TRUE;
			}
		}
		return SEXP_FALSE;
```

EDIT: I see this was described in #2685 and had a commit to fix it in August. Unfortunately it is still occurring in Nightly 20201014

**Comment by Goober5000** [comment]
This was confirmed fixed at the time.  I guess something new broke it.

**Comment by PhantomHoover** [comment]
I don't know who confirmed it but I just built the fix version and it made no difference. Suspect this bug has been outstanding since the multilock merge based on what's said in the old issue report.

**Comment by PhantomHoover** [comment]
Resolved by #2830
